CMakeLists.txt

```diff
@@ -171,39 +173,49 @@ function(add_valgrind_test NAME_WITH_PREFIX BINARY WORKING_DIRECTORY)
     endif()
 endfunction()
 
-get_target_property(QMAKE Qt::qmake IMPORTED_LOCATION)
-set(CARGO_ENV "QMAKE=set:${QMAKE}")
-set(RUNTIME_ENV "")
-
-# On windows, Qt dll needs to be in the PATH for the tests to run
-if(CMAKE_SYSTEM_NAME STREQUAL "Windows")
-    execute_process(
-        COMMAND ${QMAKE} -query QT_INSTALL_BINS
-        OUTPUT_VARIABLE QT_INSTALL_BINS
-        OUTPUT_STRIP_TRAILING_WHITESPACE
-    )
-
-    execute_process(
-        COMMAND ${QMAKE} -query QT_INSTALL_PLUGINS
-        OUTPUT_VARIABLE QT_INSTALL_PLUGINS
-        OUTPUT_STRIP_TRAILING_WHITESPACE
-    )
-
-    execute_process(
-        COMMAND ${QMAKE} -query QT_INSTALL_QML
-        OUTPUT_VARIABLE QT_INSTALL_QML
-        OUTPUT_STRIP_TRAILING_WHITESPACE
-    )
-
-    list(
-        APPEND
-        RUNTIME_ENV
-        "PATH=path_list_append:${QT_INSTALL_BINS}"
-        "QT_PLUGIN_PATH=path_list_append:${QT_INSTALL_PLUGINS}"
-        "QML_IMPORT_PATH=path_list_append:${QT_INSTALL_QML}"
-        "QML2_IMPORT_PATH=path_list_append:${QT_INSTALL_QML}"
-    )
-    list(APPEND CARGO_ENV ${RUNTIME_ENV})
+if(BUILD_WASM)
+    set(Rust_CARGO_TARGET wasm32-unknown-emscripten)
```

Comment:
can we try setting this on the CLI, as note some people try to build the examples standalone, eg they run CMake from the examples folder. So we either need to have this in all the examples or instead maybe it's better in the page in the book about WASM to state they need to run with the following or similar?

```
cmake -DRust_CARGO_TARGET=wasm32-unknown-emscripten
```